wlauto/utils/cros_sdk.py

```diff
@@ -0,0 +1,123 @@
+#    Copyright 2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import sys
+import subprocess
+import time
+
+from Queue import Queue, Empty
+from threading import Thread
+from subprocess import Popen, PIPE
+from wlauto.utils.misc import which
+from wlauto.exceptions import HostError
+
+
+class OutputPollingThread(Thread):
+
+    def __init__(self, out, queue, name):
+        super(OutputPollingThread, self).__init__()
+        self.out = out
+        self.queue = queue
+        self.stop_signal = False
+        self.name = name
+
+    def run(self):
+        for line in iter(self.out.readline, ''):
+            if self.stop_signal:
+                break
+            self.queue.put(line)
+
+    def set_stop(self):
+        self.stop_signal = True
+
+
+class CrosSdkSession(object):
+
+    def __init__(self, cros_path, password=''):
+        self.in_chroot = True if which('dut-control') else False
+        ON_POSIX = 'posix' in sys.builtin_module_names
+        if self.in_chroot:
+            self.cros_sdk_session = Popen(['/bin/sh'], bufsize=1, stdin=PIPE, stdout=PIPE, stderr=PIPE,
+                                          cwd=cros_path, close_fds=ON_POSIX, shell=True)
+        else:
+            cros_sdk_bin_path = which('cros_sdk')
+            if not cros_sdk_bin_path:
+                raise HostError("Failed to locate 'cros_sdk' make sure it is in your PATH")
```

Comment:
you don't need to have `cros_sdk` in your PATH, you can find it in `chromium/tools/depot_tools`.  If `which` fails, you could try to locate it there and if it's an executable, use it.